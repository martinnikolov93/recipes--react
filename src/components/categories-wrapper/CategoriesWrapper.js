import React from 'react';
import Category from '../category/Category';
import { Link } from 'react-router-dom';

const CategoriesWrapper = (props) => {
    if (!props.data) {
        return 'Loading..'
    }

    return (
        <>
            {props.data.map((category, i) => {
                return (
                    <div key={i}>
                        <h2><Link to={`categories/${category.title}`} >{category.title}</Link></h2>
                        <Category categoryTitle={category.title} limit={4} />
                    </div>
                )
            })}
        </>
    )
}

export default CategoriesWrapper