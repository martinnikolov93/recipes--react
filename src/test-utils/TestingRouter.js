import React from 'react';
import UserContext from '../Context';
import { BrowserRouter, Route } from 'react-router-dom'

const TestingRouter = ({user, children}) => {
    return (
        <BrowserRouter>
            <Route>
                <UserContext.Provider value={user}>
                    {children}
                </UserContext.Provider>
            </Route>
        </BrowserRouter>
    )
}

export default TestingRouter