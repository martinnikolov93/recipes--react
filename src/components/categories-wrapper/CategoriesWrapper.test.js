import React from 'react';
import renderer from 'react-test-renderer'
import CategoriesWrapper from './CategoriesWrapper';
import TestingRouter from '../../test-utils/TestingRouter';

jest.mock('../../components/category/Category.js', () => 'Category')

const categoriesDummy = [
    { title: 'category1' },
    { title: 'category2' },
    { title: 'category3' },
    { title: 'category4' },
]

describe('Categories dropdown component', () => {
    it('Should render Categories dropdown component', () => {
        const tree = renderer.create(
            <TestingRouter>
                <CategoriesWrapper data={categoriesDummy} />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})