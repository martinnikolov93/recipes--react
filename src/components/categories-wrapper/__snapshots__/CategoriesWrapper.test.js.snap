// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories dropdown component Should render Categories dropdown component 1`] = `
Array [
  <div>
    <h2>
      category1
    </h2>
    <Category
      categoryTitle="category1"
      limit={4}
    />
  </div>,
  <div>
    <h2>
      category2
    </h2>
    <Category
      categoryTitle="category2"
      limit={4}
    />
  </div>,
  <div>
    <h2>
      category3
    </h2>
    <Category
      categoryTitle="category3"
      limit={4}
    />
  </div>,
  <div>
    <h2>
      category4
    </h2>
    <Category
      categoryTitle="category4"
      limit={4}
    />
  </div>,
]
`;
